isl/ion_schema_1_0/non_negative_int.isl

```diff
@@ -18,5 +18,6 @@ type::{
     { one_of: [ range_boundary_non_negative_int, { valid_values: [min] } ] },
     { one_of: [ range_boundary_non_negative_int, { valid_values: [max] } ] },
   ],
+  not: { contains: [min, max]},
```

Comment:
There are two questions when reading this type definition and would like to get some explanation: For this type definition, we are trying to define the schema of the range of non-negative int, i.e. range::[0, 100]. From what I understand the `ordered_elements` defines the lower bound value and upper bound value of `range_non_negative_int`. I'm wondering why we would include `valid_values:[min]` in the lower bound value? Is `not: { contains: [min, max]} `trying to prohibit `range_non_negative_int` from being` range:: [min, max]` ? Thanks.

Reply:
Yes, this means that it should not contain both `min` _and_ `max`.

Reply:
> I'm wondering why we would include valid_values:[min] in the lower bound value?

It's for the user experience.

Yes, we could prohibit `min` and require people to use `0` instead. We also allow `min` for `range_positive_int` even though we know the min value is 1, and we also allow `min` for `range_timestamp_precision` even though the min precision is `year`.

By allowing `min` for all of these, it's a lower cognitive burden for people who are using Ion Schema—you don't _need_ to memorize the lower bound for every type of range. Also, the symbol `min` isn't just data, it also communicates _intention_ to future readers of the schema in a way that is more clear than just putting the min value into the range.